### PR TITLE
[docs][TypeProf]Update instrumentation file format document

### PR DESCRIPTION
**Pull request by mingmingl-llvm**
This is a follow-up to the [profile format change ](https://github.com/llvm/llvm-project/pull/82711)

### CODE REVIEW

File: llvm/docs/InstrProfileFormat.rst [hunk 1]
```diff
@@ -333,6 +342,37 @@ Function names serve as keys in the PGO data hash table when raw profiles are
 converted into indexed profiles. They are also crucial for ``llvm-profdata`` to
 show the profiles in a human-readable way.
 
+Virtual Table Profile Data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section is used for `type profiling`_. Each entry corresponds to one virtual
+table and could be illustrated with the following C++ struct
```
[the remark this answers is not in the document]
Comment (minglotus-6) [reply llvm/docs/InstrProfileFormat.rst:349]: done.

File: llvm/docs/InstrProfileFormat.rst [hunk 2]
```diff
@@ -333,6 +342,37 @@ Function names serve as keys in the PGO data hash table when raw profiles are
...
+  struct VTableProfileInfo {
+    // The start address of the vtable, collected at runtime.
+    uint64_t StartAddress;
+    // The byte size of the vtable. `StartAddress` and `ByteSize` specifies an address range to look up.
+    uint32_t ByteSize;
+    // The hash of vtable's (PGO) name
+    uint64_t MD5HashOfName;
+  };
+
+At profile use time, compiler looks up a profiled address in the sorted vtable
```
[the remark this answers is not in the document]
Comment (minglotus-6) [reply llvm/docs/InstrProfileFormat.rst:362]: done.

File: llvm/docs/InstrProfileFormat.rst [hunk 3]
```diff
@@ -333,6 +342,37 @@ Function names serve as keys in the PGO data hash table when raw profiles are
...
+address ranges and maps the address to a specific vtable through hashed name.
+
+Virtual Table Names
+^^^^^^^^^^^^^^^^^^^^
+
+This section is similar to `function names`_ section above, except it contains the PGO
+names of profiled virtual tables. It's a standalone section such that raw profile
+readers could directly find each name set by accessing the corresponding profile
+data seciton (i.e., no join across sections to tell vtable from functions).
```
[the remark this answers is not in the document]
Comment (minglotus-6) [reply llvm/docs/InstrProfileFormat.rst:371]: done.

[the remark this answers is not in the document]
Comment (minglotus-6) [reply llvm/docs/InstrProfileFormat.rst:371]: done.

File: llvm/docs/InstrProfileFormat.rst [hunk 4]
```diff
@@ -455,6 +497,16 @@ Temporal Profile Traces
 The section is used to carry on temporal profile information from raw profiles.
 See `temporal profiling`_ for the design.
 
+Virtual Table Names
+^^^^^^^^^^^^^^^^^^^^
+This section is used to carry on the names of vtables from raw profiles into
```
[the remark this answers is not in the document]
Comment (minglotus-6) [reply llvm/docs/InstrProfileFormat.rst:502]: done.

File: llvm/docs/InstrProfileFormat.rst [hunk 5]
```diff
@@ -455,6 +497,16 @@ Temporal Profile Traces
...
+indexed profiles.
+
+Unlike function names which are stored as keys of `function data`_ hash table,
+vtable names needs to be stored in a standalone section in indexed profiles.
```
[the remark this answers is not in the document]
Comment (minglotus-6) [reply llvm/docs/InstrProfileFormat.rst:506]: done.

**Review by david-xl — APPROVED** [review]

**Comment by minglotus-6** [comment]
thanks for the review! On a second look I found this PR doesn't update ASCII art and header for indexed profiles.

I will send out a separate PR to make indexed profile ASCII art and header more general and high-level -> pointing users to look at source code to know the exact fields, and document the high-level summary that offsets are used to indicate the position of sections.

**Pull request merged**